src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.ts

```diff
@@ -109,52 +115,48 @@ export class NvmeofNamespacesFormComponent implements OnInit {
 
   createForm() {
     this.nsForm = new CdFormGroup({
-      image: new UntypedFormControl(`nvme_ns_image:${Date.now()}`, {
-        validators: [Validators.required, Validators.pattern(/^[^@/]+?$/)]
-      }),
       pool: new UntypedFormControl(null, {
         validators: [Validators.required]
       }),
       image_size: new UntypedFormControl(1, [CdValidators.number(false), Validators.min(1)]),
-      unit: new UntypedFormControl(this.units[2])
+      unit: new UntypedFormControl(this.units[2]),
+      nsCount: new UntypedFormControl(this.MAX_NAMESPACE_CREATE)
     });
   }
 
-  updateRequest(rbdImageSize: number): NamespaceUpdateRequest {
+  buildUpdateRequest(rbdImageSize: number): Observable<HttpResponse<Object>> {
     const request: NamespaceUpdateRequest = {
       gw_group: this.group,
       rbd_image_size: rbdImageSize
     };
-    return request;
+    return this.nvmeofService.updateNamespace(
+      this.subsystemNQN,
+      this.nsid,
+      request as NamespaceUpdateRequest
+    );
   }
 
-  createRequest(rbdImageSize: number): NamespaceCreateRequest {
-    const image = this.nsForm.getValue('image');
-    const pool = this.nsForm.getValue('pool');
-    const request: NamespaceCreateRequest = {
-      gw_group: this.group,
-      rbd_image_name: image,
-      rbd_pool: pool
-    };
-    if (rbdImageSize) {
-      request['rbd_image_size'] = rbdImageSize;
-    }
-    return request;
+  randomString() {
```

Comment:
should we simply create a random string or something that makes sense for user? because if I opt in to create 5 namespaces I'd like to find them easier in the UI so that atleast those 5 names should be clearly mentioned in the create form before I go on and submit the form. One thing I have seen in the vSphere plugin is that they take the name like `namespace_test`, mention in the textbox that it'll create namespaces like below and then it goes on to create 
```
namespace_test_1
namespace_test_2
namespace_test_3
namespace_test_4
namespace_test_5
```

Reply:
Actually this is the name we are following :
https://github.com/ceph/ceph/blob/ac43e5353b7cf0b8a920788a7472507f20066c53/src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.ts#L150

Reply:
even that is lost for the user right since if they are dealing with many number of namespaces and they want to search for a specific namespace by name where something he can recognize. Even if we are going with this name, I'd display that format on the form so he can atleast follow on the name later on.

Reply:
yes, I think user is mostly on this page concerned with creating namespace and there will be nsid whcih will increase. I  think listing names would be nice there should be some way user can know that these are new ones IMO

**Future**:

 And I think in UI we should have some way to highlight newly added items in table like for some few seconds perhaps. 

I would say we may less care of the image here but its important and ideally we should have some linking to the page. 

**Short term**

Lets add an already checked checkbox saying want to add image names ?
or provide image name filed right away pre-populated
or just list the 5 namesapces

lets sync and talk on this also can involve UX and QE

Thanks

Reply:
https://tracker.ceph.com/issues/70004